opensrp-reveal/src/main/java/org/smartregister/reveal/view/ListTasksActivity.java

```diff
@@ -223,9 +225,26 @@ public void onDrawerStateChanged(int newState) {//do nothing
 
         NavigationView navigationView = findViewById(R.id.nav_view);
         View headerView = navigationView.getHeaderView(0);
-        int screenHeightPixels = getResources().getDisplayMetrics().heightPixels
-                - getResources().getDimensionPixelSize(R.dimen.drawer_separator_margin);
-        headerView.setLayoutParams(new LayoutParams(LayoutParams.MATCH_PARENT, screenHeightPixels));
+        headerView.getViewTreeObserver().addOnGlobalLayoutListener(new ViewTreeObserver.OnGlobalLayoutListener() {
+            @Override
+            public void onGlobalLayout() {
+                headerView.getViewTreeObserver().removeOnGlobalLayoutListener(this);
+                //if content of hamburger menu is bigger than screen; scroll content
+                if (getResources().getDisplayMetrics().heightPixels
+                        < headerView.getHeight() + getResources().getDimensionPixelSize(R.dimen.hamburger_margin)) {
+                    headerView.setLayoutParams(new LayoutParams(LayoutParams.MATCH_PARENT, LayoutParams.MATCH_PARENT));
+                    View operator = headerView.findViewById(R.id.operator_label);
+                    ConstraintLayout.LayoutParams params = (ConstraintLayout.LayoutParams) operator.getLayoutParams();
+                    params.height = ConstraintLayout.LayoutParams.WRAP_CONTENT;
+                    operator.setLayoutParams(params);
+                } else {//content of hamburger menu fits on screen; set menu height to screen height
+                    int screenHeightPixels = getResources().getDisplayMetrics().heightPixels
+                            - getResources().getDimensionPixelSize(R.dimen.hamburger_margin);
```

Comment:
define `getResources().getDisplayMetrics().heightPixels` and ` getResources().getDimensionPixelSize(R.dimen.hamburger_margin)` at the top of the function as they're used in both conditionals.

Reply:
done